commit df58ef51caea073f4b45857c51f445f65267dac7
Author: Francois Romieu <romieu@fr.zoreil.com>
Date:   Thu Oct 9 14:35:58 2008 -0700

    r8169: fix early spinlock use
    
    rtl8169_init_one
    -> rtl_init_mac_address
       -> rtl_rar_set
          -> spin_lock_irq(&tp->lock);
    [...]
    -> spin_lock_init(&tp->lock);
    
    Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 108aaff..59fbc64 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -1952,7 +1952,8 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 		dev->do_ioctl = rtl8169_ioctl;
 	}
 
-	/* Read MAC address from EEPROM */
+	spin_lock_init(&tp->lock);
+
 	rtl_init_mac_address(tp, ioaddr);
 
 	/* Get MAC address */
@@ -1995,8 +1996,6 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 	tp->timer.data = (unsigned long) dev;
 	tp->timer.function = rtl8169_phy_timer;
 
-	spin_lock_init(&tp->lock);
-
 	rc = register_netdev(dev);
 	if (rc < 0)
 		goto err_out_msi_5;
